casanovo/denovo/model_runner.py

```
@@ -520,6 +520,19 @@ def initialize_model(self, train: bool, db_search: bool = False) -> None:
+
+            # the check and raise error part
+            model_vocab_size = self.model.vocab_size
+            tokenizer_vocab_size = len(tokenizer) + 1
+            if model_vocab_size > tokenizer_vocab_size:
+                raise ValueError(
+                    f"The model was trained with a vocabulary of size "
+                    f"{model_vocab_size}, but the current config only "
+                    f"defines a vocabulary of size {tokenizer_vocab_size}. "
+                    f"Ensure your config's `residues` match those used "
+                    f"during training."
+                )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)